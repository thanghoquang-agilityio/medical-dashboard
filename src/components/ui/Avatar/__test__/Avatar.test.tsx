import { render, screen } from '@testing-library/react';

import { Avatar } from '..';

describe('Avatar Component', () => {
  it('renders with default size (md) and without border', () => {
    const { container } = render(<Avatar />);

    expect(container).toMatchSnapshot();
  });

  it('renders with border when isBordered is true', () => {
    render(<Avatar isBordered />);

    const spanElement = screen.getByTestId('avatar-container');

    // Check for bordered class
    expect(spanElement).toHaveClass('ring-offset-2 ring-1 ring-green');
  });

  it('renders with border when isCustomBordered is true', () => {
    render(<Avatar isCustomBordered />);

    const spanElement: HTMLSpanElement = screen.getByTestId('avatar-container');

    // Check for bordered class
    expect(spanElement).toHaveClass(
      'ring-offset-0 ring-warning ring-opacity-25 ring-4',
    );
  });
});
