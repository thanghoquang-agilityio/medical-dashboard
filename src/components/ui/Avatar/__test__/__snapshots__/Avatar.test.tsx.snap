// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar Component renders with default size (md) and without border 1`] = `
<div>
  <span
    class="box-border overflow-hidden outline-none rounded-full ring-offset-background dark:ring-offset-background-dark shrink-0 min-w-8 w-8 h-8 bg-undefined ring-offset-0 ring-0 false"
    data-testid="avatar-container"
    tabindex="-1"
  >
    <img
      alt=""
      data-nimg="1"
      decoding="async"
      fetchpriority="high"
      height="48"
      src="http://localhost/_next/image?url=%2Fimages%2Favatar-thumbnail.webp&w=96&q=75"
      srcset="/_next/image?url=%2Fimages%2Favatar-thumbnail.webp&w=48&q=75 1x, /_next/image?url=%2Fimages%2Favatar-thumbnail.webp&w=96&q=75 2x"
      style="color: transparent; object-fit: cover; background-size: cover; background-position: 50% 50%; background-repeat: no-repeat;"
      width="48"
    />
  </span>
</div>
`;
